### PR TITLE
Allow multi select

### PR DESCRIPTION
**Pull request by deanbarrow**
This PR allows you to `Shift+Enter` to select multiple items to insert into the clipboard.

Useful for copying multiple items from a page and pasting them together in one go.

Credit to https://unix.stackexchange.com/a/140729 for `awk 'NR>1{print PREV} {PREV=$0} END{printf("%s",$0)}'`

### CODE REVIEW

File: clipmenu [hunk 1]
```diff
@@ -57,11 +57,25 @@ else
...
+
+files=()
+while IFS= read -r line; do
+    file=$cache_dir/$(cksum <<< "$line")
+    files+=("$file")
+done <<< "$chosen_line"
+
+paste=()
+for file in "${files[@]}"; do
+    contents=$(cat "$file")
```
Comment (cdown) [inline clipmenu:69]: Command substitution mangles cases with trailing newlines -- take a look at how this was covered in clipmenud for xsel.

File: clipmenu [hunk 2]
```diff
@@ -57,11 +57,25 @@ else
...
+    paste+=("$contents")
```
Comment (cdown) [inline clipmenu:70]: Why do all this, why not actually use `cat` to concat and send it directly to xsel?

File: clipmenu [hunk 3]
```diff
@@ -57,11 +57,25 @@ else
...
+done
+
+paste=$( IFS=$'\n'; echo "${paste[*]}" )
+
+ [[ $paste ]] || exit 2
 
 for selection in clipboard primary; do
-    xsel --logfile /dev/null -i --"$selection" < "$file"
+    echo "$paste" | awk 'NR>1{print PREV} {PREV=$0} END{printf("%s",$0)}' | xsel --logfile /dev/null -i --"$selection"
```
Comment (cdown) [inline clipmenu:78]: Wait, why are you explicitly mangling trailing newlines? That's literally something we try as hard as possible to preserve, so doing that is a non-starter.

File: clipmenu [hunk 4]
```diff
@@ -57,11 +57,25 @@ else
...
+    files+=("$file")
+done <<< "$chosen_line"
+
+paste=()
+for file in "${files[@]}"; do
+    contents=$(cat "$file")
+    paste+=("$contents")
+done
+
+paste=$( IFS=$'\n'; echo "${paste[*]}" )
```
Comment (cdown) [inline clipmenu:73]: This just replaces `${paste[0]}` and leaves everything else intact. Either way, it seems unnecessary -- just use cat to join the data.

**Review by cdown — CHANGES_REQUESTED** [review]
Hi! I'm afraid I'm a bit confused by what you're trying to achieve here. Could you please clarify your intentions?

File: clipmenu (hunk 2 above)
Comment (deanbarrow) [reply clipmenu:70]: I tried with cat but it didn't work, I'll have another go as it would simplify the code a lot.

File: clipmenu (hunk 3 above)
Comment (deanbarrow) [reply clipmenu:78]: This was specifically to remove the trailing newline that `echo` creates. I'm not sure if `cat` would behave the same but I'll test it.

File: clipmenu (hunk 4 above)
Comment (deanbarrow) [reply clipmenu:73]: Will take a look.

File: clipmenu (hunk 1 above)
Comment (deanbarrow) [reply clipmenu:69]: Will check, thanks.

**Comment by cdown** [comment]
Closing without prejudice due to lack of update, feel free to resubmit. Thanks.

**Pull request closed without merging**